packages/docusaurus-cssnano-preset/src/remove-overridden-custom-properties/index.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const postcss = require('postcss');
+
+/*
+This PostCSS plugin will remove duplicate/same custom properties (which are actually overridden ones) **only** from `:root` selector.
+
+Depending on the presence of an `!important` rule in value of custom property, the following actions will happens:
+
+- If the same custom properties do **not** have an `!important` rule, then all of them will be removed except for the last one (which will actually be applied).
+- If the same custom properties have at least one `!important` rule, then only those properties that do not have this rule will be removed.
+*/
+
+module.exports = postcss.plugin(
```

Comment:
Isn't it the goal of purgecss to remove such unused css?

I'd rather add a widely used tool for this than maintain our own, but I don't know enough purgecss to be sure it works for this specific usecase

Reply:
purgecss does a very different job - it removes unused from HTML code CSS classes. Currently we are improving minification (optimizing/reducing CSS code), by merging rules, selectors, removing unnecessary CSS vars, and much more.

But in order to achieve an even smaller CSS bundle, we should still try to integrate purgecss into the minification process. However I don’t think this is an easy task, so I started with an easier path.

Reply:
I see make sense :)